tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+        plot_bounds = ogr.CreateGeometryFromJson(str(geometry))
+        bad_result = geoimage.clip_raster_intersection(TEST_IMAGE, file_bounds, plot_bounds,
+                                                       "images/geoimage_clip_raster_intersection.tif")
+        assert bad_result is None
+    complete_overlap = str(geoimage.clip_raster_intersection(TEST_IMAGE, file_bounds, file_bounds,
+                                                             "images/geoimage_clip_raster_intersection.tif"))
+    with open("data/geoimage_clip_raster_intersection.txt", 'r') as checkfile:
+        checkfile_content = checkfile.read()
+        assert checkfile_content == complete_overlap
+
+
+def test_geoimage_clip_raster_intersection_json():
+    """Checks the output when calling clip_raster_intersection_json on an image, usingmetadata from the
+    data folder in order to make sure that the function output is correct"""
+    file_bounds = geoimage.get_image_bounds(TEST_IMAGE)
+    json_file_bounds = geometries.geometry_to_geojson(file_bounds)
+    geo_json = json.load(open(JSON_FILE))
+    for feature in range(len(geo_json["features"])):
+        json_plot_bounds = geo_json["features"][feature]["geometry"]
+        bad_result = geoimage.clip_raster_intersection_json(TEST_IMAGE, json_file_bounds, json_plot_bounds,
+                                                            "images/geoimage_clip_raster_intersection.tif")
+        assert bad_result is None
+    complete_overlap = geoimage.clip_raster_intersection_json(TEST_IMAGE, json_file_bounds, json_file_bounds,
+                                                              "images/geoimage_clip_raster_intersection.tif")
+    with open("data/geoimage_clip_raster_intersection.txt", 'r') as checkfile:
+        checkfile_content = checkfile.read()
+        assert checkfile_content == str(complete_overlap)
+
+
+def test_geoimage_create_geotiff():
+    """Tests create_geotiff, although a complete image is not generated at the moment. gps_bounds were extracted
+    from output.tin.tif"""
+    gps_bounds = (3971937.000, 3972667.000, 368908.000, 369799.000)
+    data = gdal.Open(TEST_IMAGE)
+    myarray = np.array(data.GetRasterBand(1).ReadAsArray())
+    if os.path.isfile("images/output.tif"):
+        os.remove("images/output.tif")
+    geoimage.create_geotiff(myarray, gps_bounds, "images/output.tif", 26913)
+
+
+def test_geoimage_get_epsg():
+    """Test to see whether get_epsg correctly locates the epsg from a geotiff image"""
+    epsg = geoimage.get_epsg(TEST_IMAGE)
+    assert epsg == "26913"
+
+
+# I don't know how to test an input where the file does not have an epsg/epsg = None
+def test_geoimage_get_image_bounds():
+    """Tests get_image_bounds on a geotiff image and makes sure that the correct polygons are returned"""
+    bad_file_res = geoimage.get_image_bounds("")
+    assert bad_file_res is None
+    res = geoimage.get_image_bounds(TEST_IMAGE)
```

Comment:
It would be better to have the image name and expected string in a JSON file so that it could be looped through (and extended easily)

Reply:
Why should the image name be included in the JSON file rather than as a constant? I understand the part regarding putting the expected output in the JSON file and have made that happen.

Reply:
It would allow different geo referenced images to be easily specified along with their expected results. For example, to test out an image that uses a different way of storing coordinate information. Or, perhaps there's a coordinate system that's not installed by default, and should be - it would allow that test to take place. No problem with the way things are done now if you don't want to change it